lib/elixir/lib/dict.ex

```diff
@@ -107,6 +107,9 @@ defmodule Dict do
   @type value :: any
   @type t :: list | map
 
+  @type env :: t
+  defmacrocallback new2(env) :: Macro.t
```

Comment:
Here is how it will look like for `Dict` module:
```
iex(1)> b Dict
defcallback delete(t(), key()) :: t()
defcallback drop(t(), Enum.t()) :: t()
defcallback equal?(t(), t()) :: boolean()
defcallback fetch(t(), key()) :: {:ok, value()} | :error
defcallback fetch!(t(), key()) :: value() | no_return()
defcallback get(t(), key()) :: value()
defcallback get(t(), key(), value()) :: value()
defcallback get_and_update(t(), key(), (value() -> {value(), value()})) :: {value(), t()}
defcallback get_lazy(t(), key(), (() -> value())) :: value()
defcallback has_key?(t(), key()) :: boolean()
defcallback keys(t()) :: [key()]
defcallback merge(t(), t()) :: t()
defcallback merge(t(), t(), (key(), value(), value() -> value())) :: t()
defcallback new() :: t()
defmacrocallback new2(env()) :: Macro.t()
defcallback pop(t(), key()) :: {value(), t()}
defcallback pop(t(), key(), value()) :: {value(), t()}
defcallback pop_lazy(t(), key(), (() -> value())) :: {value(), t()}
defcallback put(t(), key(), value()) :: t()
defcallback put_new(t(), key(), value()) :: t()
defcallback put_new_lazy(t(), key(), (() -> value())) :: t()
defcallback size(t()) :: non_neg_integer()
defcallback split(t(), Enum.t()) :: {t(), t()}
defcallback take(t(), Enum.t()) :: t()
defcallback to_list(t()) :: list()
defcallback update(t(), key(), value(), (value() -> value())) :: t()
defcallback update!(t(), key(), (value() -> value())) :: t() | no_return()
defcallback values(t()) :: [value()]
```